### DISCUSSION TITLE
Confused on create-img :url-src

### DISCUSSION THREAD

**Discussion by sabracrolleton** [rabbibotton/clog#160] [Q&A] [answered]
So far I have been unable to give create-img a :url-src filename that is outside the static route and have it work. 

My current process is to override the static-route in the initialization (pointing to a directory in my own project), copying all the necessary files. I then generate svg diagrams in a sub-directory of that new static route. Then if a user wants to load a diagram, they can use server-file-dialog to find it, but then I have to drop everything in the returned pathname except the file-namestring.

It works, but I feel like I am missing something obvious. (Then I need to figure out how to add zoom in and out to the image view - the svg diagrams get large).

**Comment by rabbibotton** [comment, marked as answer]
The issue is that the :url-src has to be served from a webserver not a local file it is a security limitation. You can try using a file URL file:///Users/dbotton/Desktop/advanced%20format.pdf if your app is run on 127.0.0.1 it should work.

**Reply by sabracrolleton** [reply]
Thank you for the explanation

**Reply by sabracrolleton** [reply]
Hmm. The suggestion to use a file URL did not seem to work.

**Reply by rabbibotton** [reply]
browser security policies often create big issues with local files directly, I tend to find ways to pass the files through the webserver to avoid issues.

**Comment by rabbibotton** [comment]
BTW  (directory-namestring path) will strip out just the dir portion of path
